### PR TITLE
NIAD-2394 - Refactor to remove dependency on the JSON bundle when building a skeleton EHR Extract

### PR DESCRIPTION
**Pull request by MartinWheelerMT**
## What

Refactor to remove dependency on the JSON bundle when building a skeleton EHR Extract
Add Unit test and resource files to test the modified method is working as expected

## Why

This is the first part of refactoring to prepare for changing when the bundle generation occurs.

## Type of change

Please delete options that are not relevant.

- [x] Internal change (non-breaking change with no effect on the functionality affecting end users)

## Checklist:

- [ x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapper.java [hunk 1]
```diff
@@ -66,25 +70,22 @@ public EhrExtractTemplateParameters mapBundleToEhrFhirExtractParams(
...
+        var end = realEhrExtract.substring(endTagIndex);
+
+        return start + ehrCompositionWithNarrativeStatement + end;
     }
 
     public String buildEhrCompositionForSkeletonEhrExtract(String documentId) {
         var skeletonComponentTemplateParameters = SkeletonComponentTemplateParameters.builder()
             .narrativeStatementId(documentId)
             .availabilityTime(DateFormatUtil.toHl7Format(timestampService.now()))
-            .effectiveTime(StatementTimeMappingUtils.prepareEffectiveTimeForEhrFolder(messageContext.getEffectiveTime()))
```
Comment (adrianclay) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapper.java:87]: Remove the `effectiveTime` field from this class?

File: service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/StructuredRecordMappingService.java [hunk 2]
```diff
@@ -136,11 +136,8 @@ public String mapStructuredRecordToEhrExtractXml(GetGpcStructuredTaskDefinition
         return outputMessageWrapperMapper.map(structuredTaskDefinition, ehrExtractContent);
     }
 
-    public String buildSkeletonEhrExtractXml(
-        GetGpcStructuredTaskDefinition structuredTaskDefinition, Bundle structuredRecord, String documentId
+    public String buildSkeletonEhrExtractXml(String realEhrExtract, String documentId
```
Comment (adrianclay) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/StructuredRecordMappingService.java:139]: I'm not sure how I feel about this indirection. Could the `ehrExtractMapper` method be called directly instead?

File: service/src/test/resources/ehr/mapper/ehrExtract/ehrExtract.xml [hunk 3]
```diff
@@ -0,0 +1,8160 @@
+<EhrExtract classCode="EXTRACT" moodCode="EVN">
```
Comment (adrianclay) [inline service/src/test/resources/ehr/mapper/ehrExtract/ehrExtract.xml:1]: Shall we try and simplify this XML snippet?

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapper.java [hunk 4]
```diff
@@ -66,25 +79,46 @@ public EhrExtractTemplateParameters mapBundleToEhrFhirExtractParams(
...
+        var startTagIndex = realEhrExtract.indexOf(RESPONSIBLE_PARTY_END_TAG);
+        var endTagIndex = realEhrExtract.lastIndexOf(EHR_FOLDER_END_TAG);
+
+        var start = realEhrExtract.substring(0, startTagIndex + RESPONSIBLE_PARTY_END_TAG.length() + 1);
+        var end = realEhrExtract.substring(endTagIndex);
+
+        return tryFormatXml(start + ehrCompositionWithNarrativeStatement + end);
+    }
+
+    private static String tryFormatXml(String skeletonEhrExtract) {
```
[the remark this answers is not in the document]
Comment (MartinWheelerMT) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapper.java]: This has been removed as we do not care about the formatting of the XML

File: service/src/test/resources/ehr/mapper/ehrExtract/ehrExtract.xml (hunk 3 above)
Comment (MartinWheelerMT) [reply service/src/test/resources/ehr/mapper/ehrExtract/ehrExtract.xml:1]: Have removed any components which were not required for the process to complete, whilst leaving in 2 components to ensure that the test is effective

**Review by adrianclay — APPROVED** [review]
I feel like we could be a lot more heavy with the pruning sheers on the fixtures, but not a blocker.

**Pull request merged**